x-pack/apm-server/main.go

```diff
@@ -117,6 +119,24 @@ func newTailSamplingProcessor(args beater.ServerParams) (*sampling.Processor, er
 		return nil, fmt.Errorf("failed to create Elasticsearch client for tail-sampling: %w", err)
 	}
 
+	oldStorageDir := paths.Resolve(paths.Data, oldTailSamplingStorageDir)
+	if entries, err := os.ReadDir(oldStorageDir); err == nil {
```

Comment:
Do we want to log the error if the syscall to open a directory has failed here? i.e. specifically checking for `err != nil && err != io.EOF`